### PR TITLE
Teach how to run canary deployments

### PR DESCRIPTION
**Pull request by janetkuo**
This has been asked by the users several times so I'd like to document it. 

@bgrant0607 @nikhiljindal @thockin @kubernetes/docs

### CODE REVIEW

File: docs/user-guide/deployments.md [hunk 1]
```diff
@@ -409,6 +409,12 @@ nginx-deployment-3066724191   0         0         1h
 
 Note: A paused Deployment cannot be scaled at this moment, and we will add this feature in 1.3 release, see [issue #20853](https://github.com/kubernetes/kubernetes/issues/20853). You cannot rollback a paused Deployment either, and you should resume a Deployment first before doing a rollback. 
 
+## Use Cases 
+
+### Canary Deployment
+
+If you want to run multiple versions of a container using the Deployment, you can just create multiple Deployments for different versions.
```
Comment (thockin) [inline docs/user-guide/deployments.md]: Can you expand on it a little bit more?  something like "create multiple deployments, and when you're happy with one, you can update your main deployment to the new version".  Maybe talk about using a Service over multiple deployments.  Just a little more for people who can't read your mind :)

Comment (bgrant0607) [inline docs/user-guide/deployments.md]: I wouldn't put this here. I'd expand:
http://kubernetes.io/docs/user-guide/managing-deployments/#canary-deployments

and maybe link to an example, such as:
https://github.com/kelseyhightower/talks/tree/master/kubecon-eu-2016/demo#deploy-a-canary

**Comment by janetkuo** [comment]
@bgrant0607 @thockin Comments addressed the published at 
- http://janetkuo.github.io/docs/user-guide/deployments/#canary-deployment
- http://janetkuo.github.io/docs/user-guide/managing-deployments/#canary-deployments

PTAL

File: docs/user-guide/deployments.md (hunk 1 above)
Comment (janetkuo) [inline docs/user-guide/deployments.md]: Done

**Comment by janetkuo** [comment]
@bgrant0607 PTAL

**Comment by bgrant0607** [comment]
LGTM

**Pull request merged**